lib/cdo/poste.rb

```diff
@@ -110,41 +110,70 @@ def self.unsubscribe(email, hashed_email, params={})
   end
 
   class Template
-    def initialize(body, engine=TextRender::MarkdownEngine)
+    def initialize(path)
+      body = IO.read(path)
+
+      @engine = {
+        '.haml' => TextRender::HamlEngine,
+        '.html' => TextRender::ErbEngine,
+        '.md' => TextRender::MarkdownEngine,
+      }[File.extname(path).downcase]
+
       if match = body.match(/^---\s*\n(?<header>.*?\n?)^(---\s*$\n?)(?<html>\s*\n.*?\n?)^(---\s*$\n?)(?<text>\s*\n.*?\n?\z)/m)
-        @header = TextRender::YamlEngine.new(match[:header].strip)
-        @html = engine.new(match[:html].strip)
-        @text = TextRender::ErbEngine.new(match[:text].strip)
+        @header = match[:header].strip
+        @html = match[:html].strip
+        @text = match[:text].strip
       elsif match = body.match(/^---\s*\n(?<header>.*?\n?)^(---\s*$\n?)(?<html>\s*\n.*?\n?\z)/m)
-        @header = TextRender::YamlEngine.new(match[:header].strip)
-        @html = engine.new(match[:html].strip)
+        @header = match[:header].strip
+        @html = match[:html].strip
       else
-        @html = engine.new(body.strip)
+        @html = body.strip
       end
     end
 
     def render(params={})
       if params.key?('form_id')
-        form = Form2.from_row(DB[:forms].where(id: params['form_id']).first)
+        form = Form2.from_row(POSTE_DB[:forms].where(id: params['form_id']).first)
         params.merge! form.data
         params.merge! form.processed_data
         params['form'] = form
       end
       locals = OpenStruct.new(params).instance_eval {binding}
 
-      header = @header.result(locals) unless @header.nil?
+      header = render_header(locals)
+
       # TODO(andrew): Fix this so that we get a signal as to how often this is happening.
       # For more information, see https://www.pivotaltracker.com/story/show/104750788.
```

Comment:
Looks like you've removed a guard clause here (hopefully unnecessary now).  Should this TODO also be removed? I'm pretty sure that's what it was referring to.

Reply:
oh, good call. I don't have insight into whether or not the TODO would still be useful; do you have context for our use of this litmus tracking feature, or do you know who might?

Reply:
:joy: Maybe?  Looks like it was introduced in November 2014 by @geoffrey-elliott in [this commit named "Fix conflicts."](https://github.com/code-dot-org/code-dot-org/commit/19e5aa707d01614ae2473b092fd19629eae03439#diff-fe80c20e87b389e81c85fe8da32b3afcR75) (If he's watching GitHub maybe he can shed some light on the situation.)

I checked out `emltrk.com` (which was blocked by uBlock, by the way) and Litmus does seem to still be a working service.  I have credentials in LastPass.  It looks like we've been on a "free account" for more than three years now, which doesn't have any features except for the ability to start paying again.

It also looks like we've been using the same tracking id in our HOC emails for years.

I think we can safely tear this out.  Might check with #marketing _just_ to be sure first.

Reply:
Thanks for the sleuthing!

![4136b77b16937f3c47c061b9f1625489](https://user-images.githubusercontent.com/244100/75714222-27b58c00-5c80-11ea-9181-d61ed6d0e10e.gif)

I'll take out this TODO and add a work item to track tearing out the tracking logic

Reply:
Addressing in https://github.com/code-dot-org/code-dot-org/pull/33402.